### DISCUSSION TITLE
GTK: iBus IME for unicode and emoji input doesn't work and breaks ctrl+shift+u keybind

### DISCUSSION THREAD

**Discussion by Diego-0110** [ghostty-org/ghostty#3762] [Feature Requests, Ideas]
Trying to use `ctr+shift+u` as a trigger I found that this combination of keys is the default keybind to start unicode input from iBus, so I change the default from `ibus-setup`. But after that, pressing `ctrl+shift+u` causes the keybinds stop working until you press a normal key like `a` or `f`.

[ibus_keybind_bug.webm](https://github.com/user-attachments/assets/e7dde170-257f-4295-b588-b18fb26a6bb0)

I've also tried to use (with the default keybind) iBus for emoji and unicode input and in both cases it doesn't work as expected.

[ibus_bug.webm](https://github.com/user-attachments/assets/c1181b48-4833-4de3-b177-eaf640966a04)
```
Ubuntu 24.04.1 LTS

❯ ghostty --version
Ghostty 1.0.0

Version
  - version: 1.0.0
  - channel: stable
Build Config
  - Zig version: 0.13.0
  - build mode : builtin.OptimizeMode.ReleaseFast
  - app runtime: apprt.Runtime.gtk
  - font engine: font.main.Backend.fontconfig_freetype
  - renderer   : renderer.OpenGL
  - libxev     : main.Backend.io_uring
  - GTK version:
    build      : 4.14.2
    runtime    : 4.14.2
  - libadwaita : enabled
    build      : 1.5.0
    runtime    : 1.5.0
```

**Comment by pluiedev** [comment]
Does this still occur on 1.1.2?

**Reply by Diego-0110** [reply]
Yes, I've tried it again with version 1.1.2 in Ubuntu 24.04 (with Wayland and with X11) and the same happens.

**Reply by pluiedev** [reply]
Which ibus version are you using? This is significant as minor versions of ibus all exhibit slightly different behavior..

**Reply by Diego-0110** [reply]
Ubuntu 24.04 comes with ibus 1.5.29-rc2. Also, I tested in another machine with Ubuntu 24.10 which comes with ibus 1.5.30 and has the same bug.
